docs/developer/design/volume_v1.0.rst

```diff
@@ -0,0 +1,294 @@
+   volume (more details in :ref:`volume-finalization`).
+2. the volume is stuck in an unrecoverable (automatically at least) error
+   state: the operatot can't do anything here: the request is considered done
+   and won't be rescheduled.
+3. the volume doesn't have a backing **PersistentVolume** (e.g.: newly created
+   volume): the operator will deploy the volume
+   (more details in :ref:`volume-deployment`).
+4. the backing **PersistentVolume** exists: the operator will check its status
+   to update the volume's status accordingly.
+
+.. uml:: volume_v1.0-main_loop_flowchart.uml
+
+
+.. _volume-deployment:
+
+Volume deployment
+~~~~~~~~~~~~~~~~~
+
+To deploy a volume, the operator needs to prepare its storage (using Salt) and
+create a backing **PersistentVolume**.
+
+If the **Volume** object has no value in its field ``Job``, it means that the
+deployment hasn't started, thus the operator will set a finalizer on the
+**Volume** object and then start the preparation of the storage using an
+asynchronous Salt call (which gives a job ID) before rescheduling the request
+to monitor the evolution of the job.
+
+If the **Volume** object has a job ID, then the storage preparation is in
+progress and the operator will monitor it until it's over.
+If the Salt job ends with an error, the operator will move the volume into a
+failed state.
+
+Otherwise (Salt job succeeded), the operator will proceed with the
+**PersistentVolume creation** (which require an extra Salt call, synchronous
+this time, to get the volume size), taking care of putting a finalizer on the
+**PersistentVolume** (so that its lifetime is tied to the **Volume** one) and
+set the **Volume** as the owner of the created **PersistentVolume**.
+
+Once the **PersistentVolume** is successfuly created, the operator will move
+the **Volume** to the `Available` state and reschedule the request (the next
+iteration will check the health of the **PersistentVolume** just created).
+
+.. uml:: volume_v1.0-deploy_volume_flowchart.uml
+
+
+.. _volume-finalization:
+
+Volume finalization
+~~~~~~~~~~~~~~~~~~~
+
+**Volume** in state **Pending** cannot be deleted (because the operator don't
+know where they are in the creation process). In such cases, the operator will
+we reschedule the request until the volume becomes either **Failed** or
+**Available**.
+
+For volumes with no backing **PersistentVolume**, the operator will directly
+reclaim the storage on the node (using an asynchronous Salt job) and upon
+completion it will remove the **Volume** finalize to let Kubernetes delete the
+object.
+
+If there is a backing **PersistentVolume**, the operator will delete it (if
+it's not already in a terminating state) and watch for the moment when it
+becomes unused (this is done by rescheduling). Once the backing
+**PersistentVolume** becomes unused, the operator will reclaim its storage and
+remove the finalizers to let the object be deleted.
```

Comment:
singular or plural for finalizers?

Reply:
plural (we remove finalizer on both PV and Volume in this case).